core/commands/get.go

```diff
@@ -70,22 +70,24 @@ may also specify the level of compression by specifying '-l=<1-9>'.
 			return
 		}
 
-		pbnd, ok := dn.(*dag.ProtoNode)
-		if !ok {
-			res.SetError(err, cmds.ErrNormal)
+		switch dn := dn.(type) {
+		case *dag.ProtoNode:
+			size, err := dn.Size()
+			if err != nil {
+				res.SetError(err, cmds.ErrNormal)
+				return
+			}
+
+			res.SetLength(size)
+		case *dag.RawNode:
+			res.SetLength(uint64(len(dn.RawData())))
```

Comment:
This makes me think that we should have sort of Size method or better unixfs interface.